File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.js
/**
 * @param {string[]} words
 * @param {number[][]} queries
 * @return {number[]}
 */
var vowelStrings = function(words, queries) {

let MySet = new Set(['a', 'e', 'i', 'o', 'u']);

    let curr=0;
    const n =words.length;
    let prefix_sum= new Array(n).fill(0);
    for(let i=0 ;i<n;i++)
    {
        let first=words[i][0];
        let last=words[i][words[i].length-1];
        if(MySet.has(first) && MySet.has(last))
        {
            curr++;
        }
        prefix_sum[i]=curr;
    }
    console.log(prefix_sum);
    let res=[];

    for(let i=0;i<queries.length;i++)
    {

        let start=queries[i][0]-1;
        let end=queries[i][1];
        res.push(prefix_sum[end]-(prefix_sum[start]||0));

    }

    return res;


    
};

File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:

        vowels=set()
        vowels.add('a')
        vowels.add('e')
        vowels.add('i')
        vowels.add('o')
        vowels.add('u')

        prefix_sum=[0]*(len(words))
        for i in range(len(words)):
            if words[i][0] in vowels and words[i][len(words[i])-1] in vowels:
                if i>=1:
                    prefix_sum[i]=prefix_sum[i-1]+1
                else:
                    prefix_sum[i]=1
            else:
                prefix_sum[i]=prefix_sum[i-1]

        res=[]
        for u,v in queries:
            if u>=1:
                res.append(prefix_sum[v]-prefix_sum[u-1])
            else:
                res.append(prefix_sum[v])

        return res


        